File: common/src/main/java/org/sandboxpowered/silica/network/play/clientbound/HeldItemChange.java
package org.sandboxpowered.silica.network.play.clientbound;

import org.sandboxpowered.silica.network.PacketByteBuf;
import org.sandboxpowered.silica.network.PacketHandler;
import org.sandboxpowered.silica.network.PacketPlay;
import org.sandboxpowered.silica.network.PlayContext;

public class HeldItemChange implements PacketPlay {
    private byte slot;

    public HeldItemChange() {
    }

    public HeldItemChange(byte slot) {
        this.slot = slot;
    }

    @Override
    public void read(PacketByteBuf buf) {
        this.slot = buf.readByte();
    }

    @Override
    public void write(PacketByteBuf buf) {
        buf.writeByte(slot);
    }

    @Override
    public void handle(PacketHandler packetHandler, PlayContext context) {

    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/network/play/clientbound/HeldItemChange.kt
package org.sandboxpowered.silica.network.play.clientbound

import org.sandboxpowered.silica.network.PacketByteBuf
import org.sandboxpowered.silica.network.PacketHandler
import org.sandboxpowered.silica.network.PacketPlay
import org.sandboxpowered.silica.network.PlayContext

class HeldItemChange(private var slot: Byte = 0) : PacketPlay {

    override fun read(buf: PacketByteBuf) {
        slot = buf.readByte()
    }

    override fun write(buf: PacketByteBuf) {
        buf.writeByte(slot.toInt())
    }

    override fun handle(packetHandler: PacketHandler, context: PlayContext) {}
}